### PR TITLE
THM: Parallel Junction

### PR DESCRIPTION
**Pull request by andrsd**
Refs #19899

### CODE REVIEW

**Comment by andrsd** [comment]
The restriction to replicated mesh will be removed in a follow-up PR.

**Review by joshuahansel — CHANGES_REQUESTED** [review]
Create a MOOSE newsletter entry.

File: modules/thermal_hydraulics/test/tests/components/junction_one_to_one_1phase/tests [hunk 1]
```diff
@@ -8,8 +8,8 @@
       Postprocessors/rhouA_right/boundary=right_channel:in
       Postprocessors/rhouA_right/execute_on='initial timestep_end'"
     csvdiff = 'junction_one_to_one_1phase_out.csv'
-    max_parallel = 1
     ad_indexing_type = 'global'
+    mesh_mode = 'replicated'
```
[the remark this answers is not in the document]
Comment (andrsd) [reply modules/thermal_hydraulics/test/tests/components/junction_one_to_one_1phase/tests]: I will open an issue.

What is surprising me is the `junction_one_to_one/tests` file. The latest branch should not be restricting junctions, only heat transfers from heat structures.

[the remark this answers is not in the document]
Comment (andrsd) [reply modules/thermal_hydraulics/test/tests/components/junction_one_to_one_1phase/tests]: nvm, you went patch by patch and at some point there was this restriction until I removed it. *slow thinking in the morning*

[the remark this answers is not in the document]
Comment (andrsd) [reply modules/thermal_hydraulics/test/tests/components/junction_one_to_one_1phase/tests]: Issue is opened [here](https://github.com/idaholab/moose/issues/20798)

**Review by joshuahansel — APPROVED** [review]

**Pull request merged**